room_service.py

```diff
@@ -25,21 +35,50 @@ def __init__(self) -> None:
         pass
 
     def find_all(self) -> List[dict]:
-        cursor.execute(" SELECT * FROM rooms WHERE occupant = ''")
+        cursor.execute(" SELECT * FROM rooms WHERE numOccupants < maxOccupants ")
         return cursor.fetchall()
 
-    def reserve_room(self, room_id: str, student_name: str):
+    def reserve_room(self, room_id: str):
         """
             reserve room, by passing in room_id and student name
         """
-        cursor.execute(" UPDATE rooms SET occupant = :name WHERE roomNum = :room_id", {'name': student_name, 'room_id': room_id})
-        conn.commit()
+
+        # check if the room is available
+        cursor.execute(""" SELECT * FROM rooms WHERE roomID = ? """, (room_id,))
+        curr_room = cursor.fetchall()[0]
+
+        # if room is full (value of 1) then throw an err
+        # TODO: add an error throw statement
+        if curr_room[-1] == 1:
+            print("Room Full!")
+            exit(-1)
+
+        curr_room_occupants = curr_room[1]
+        curr_room_max = curr_room[2]    # current room occupant limit
+
+        # updates table if the room is full
+        if curr_room_occupants == curr_room_max:
+            cursor.execute(""" UPDATE rooms 
+                SET isFull = ? 
+                WHERE roomID = ?
+                """, (1, room_id))
+            conn.commit()
+        else:
+            # update table
+            new_curr_room_occupants = curr_room[1] + 1  # increment the num of people in current room
+            cursor.execute(""" UPDATE rooms 
+                            SET numOccupants = ? 
+                            WHERE roomID = ?
+                            """, (new_curr_room_occupants, room_id))
+            conn.commit()   
+
+            return room_id  # gives the reserver the room_id for future reference
```

Comment:
Logic looks good to me👌

We'll discuss DTO and how to use that to make your code readable